File: 0896-monotonic-array/0896-monotonic-array.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        int cnt1=0, cnt2=0;
        for(int i=1; i<nums.size(); i++){
            if(nums[i]>=nums[i-1]) cnt1++;
            if(nums[i]<=nums[i-1]) cnt2++;
        }
        return cnt1==nums.size()-1 || cnt2==nums.size()-1;
    }
};

File: 0896-monotonic-array/0896-monotonic-array.java
class Solution {
    public boolean isMonotonic(int[] nums) {
        boolean increasing = true, decreasing = true;
        for(int i=1; i<nums.length; i++){
            if(nums[i] > nums[i-1]) decreasing = false;
            else if(nums[i] < nums[i-1]) increasing = false;
            if(increasing==false && decreasing==false) return false;
        }
        return true;
    }
}